### ISSUE TITLE
Error when assigning a NamedTuple to Tuple of fixed size is confusing

### ISSUE THREAD

**Issue by viblo** [microsoft/pyright#1152]
**Describe the bug**
When using a subclass of typing.NamedTuple and assign it to a variable (or calling a function) expecting the base class (Tuple) of the same size an error is raised. The error message complains about the sizes of the tuples, even when they are the same.

**To Reproduce**
See code below

**Expected behavior**
No error would be ideal if possible, or a less confusing error.

**Screenshots or Code**
```python
from typing import NamedTuple, Tuple

class Tuple1(NamedTuple):
    a: float
    b: float

t: Tuple[float, float] = Tuple1(a=1)
# Expression of type "Tuple1" cannot be assigned to declared type "Tuple[float, float]"
#  Tuple size mismatch; expected 2 but received 2 Pylance (reportGeneralTypeIssues)
```

**VS Code extension or command-line**
VS Code, using Pyright 1.1.83 (from Pylance)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.84, which I just published. It will also be included in the next release of Pylance.

**Comment by viblo** [comment]
Thanks for the quick reply and fix!
